src/System.Data.SqlClient/src/System/Data/SqlClient/SqlParameter.cs

```diff
@@ -261,7 +261,7 @@ public string XmlSchemaCollectionDatabase
             }
             set
             {
-                bool collectionIsNull = _xmlSchemaCollection != null;
+                bool collectionIsNull = _xmlSchemaCollection is null;
                 if (collectionIsNull)
```

Comment:
Can you remove the OR condition for `collectionIsNull` from second clause (L269) ? It doesn't make sense to me. Consider a scenario, if `collectionIsNull` is `true`, and `value` is also `null` you would set database to `null` and it wouldn't fail because you intialized the collection already above, but is that intended? `value !=null` is enough to check whether the database must be set.

Also, there is no need to store `bool collectionIsNull` then, you could directly check `if (_xmlSchemaCollection is null)`

Same in other places.

Reply:
Please also move the null check and initialization code to a separate function and call that before setting values to avoid code repetition in all three places.

Reply:
Moved to a separate function and commented why it does what it does. The logic feels clunky to me but I can't find a more elegant way to express it.